# SPDX-FileCopyrightText: Cynthia Rey <cynthia@cynthia.dev>
# SPDX-License-Identifier: 0BSD
--- a/zmij-js/src/lib.rs
+++ b/zmij-js/src/lib.rs
@@ -907,13 +907,6 @@ impl FloatShuffleTable {
                         j += 1;
                     }
                 }
-                // Rust port: print out `.0` in fixed (i.e. non-scientific) representation.
-                else if slot != Self::SCI_SLOT {
-                    data[out + len] = Self::POINT_POS;
-                    len += 1;
-                    data[out + len] = Self::ZERO_POS;
-                    len += 1;
-                }
 
                 if slot == Self::SCI_SLOT {
                     let mut i = 0;
@@ -2065,10 +2058,7 @@ where
         if bin_sig == Float::SigType::from(0) {
             return unsafe {
                 *buffer = b'0';
-                // Rust port: always emit `.0` for integers.
-                *buffer.add(1) = b'.';
-                *buffer.add(2) = b'0';
-                buffer.add(3)
+                buffer.add(1)
             };
         }
 
@@ -2178,20 +2168,12 @@ where
                 let point = start.add(layout.point_pos as usize);
                 *point = b'.';
                 *buffer.add(layout.last_digit_pos[has_extra_digit as usize] as usize) = last_digit;
-                buffer = buffer.add(
+                buffer.add(
                     *layout
                         .end_pos
                         .get_unchecked(num_digits + has_extra_digit as usize - 1)
                         as usize,
-                );
-
-                // Rust port: always emit `.0` for integers.
-                if buffer == point {
-                    *buffer.add(1) = b'0';
-                    buffer.add(2)
-                } else {
-                    buffer
-                }
+                )
             };
         }
 
@@ -2210,20 +2192,12 @@ where
 
             let point = start.add(point_pos as usize);
             *point = b'.';
-            buffer = buffer.add(
+            buffer.add(
                 *layout
                     .end_pos
                     .get_unchecked(num_digits + has_extra_digit as usize - 1)
                     as usize,
-            );
-
-            // Rust port: always emit `.0` for integers.
-            if buffer == point {
-                *buffer.add(1) = b'0';
-                buffer.add(2)
-            } else {
-                buffer
-            }
+            )
         };
     }
 
